core/src/test/java/com/datastax/oss/driver/internal/core/cql/CqlRequestHandlerSpeculativeExecutionTest.java

```diff
@@ -250,16 +239,14 @@ public void should_fail_if_no_more_nodes_and_initial_execution_is_last(
       node1Behavior.setWriteSuccess();
       // do not simulate a response from node1 yet
 
-      harness.nextScheduledTask(); // Discard the timeout task
+      harness.nextScheduledTimeout(); // Discard the timeout task
 
       // Run the next scheduled task to start the speculative execution. node2 will reply with a
       // BOOTSTRAPPING error, causing a RETRY_NEXT; but the query plan is now empty so the
       // speculative execution stops.
-      ScheduledTaskCapturingEventLoop.CapturedTask<?> firstExecutionTask =
-          harness.nextScheduledTask();
-      assertThat(firstExecutionTask.getInitialDelay(TimeUnit.MILLISECONDS))
-          .isEqualTo(firstExecutionDelay);
-      firstExecutionTask.run();
+      // next sceduled timeout should be the first speculative execution. Get it and run it.
```

Comment:
Nit: sceduled => scheduled
This is repeated in other places since you copy-pasted this block.